src/uct/ib/mlx5/dv/ib_mlx5dv_md.c

```diff
@@ -1490,6 +1504,16 @@ uct_ib_mlx5_devx_mem_dereg(uct_md_h uct_md,
         return status;
     }
 
+    /* If requested invalidation of only rkeys, then return success */
+    flags = UCT_MD_MEM_DEREG_FIELD_VALUE(params, flags, FIELD_FLAGS, 0);
+    if (flags & UCT_MD_MEM_DEREG_FLAG_INVALIDATE_RKEY_ONLY) {
```

Comment:
can't we always do invalidation of just rkey without introducing a new flag? Why do we need two flags now?

Reply:
IMO that would be a good idea in the future, to have just one flag
But currently there are 3 workflows which I'm aware of:
- rcache invalidates its own memory handles (e.g. during merge operation)
- invalidation of user memory handle during RMA error handling: here we don't use mpool and just invalidate and deallocate the entire memh through rcache. Maybe it can be revised later
- invalidation of RNDV PPLN fragment mpool entries

Currently this PR addresses only the last workflow, while keeping behaviour of the two first workflows unmodified. So the idea was to continue using `INVALIDATE` flag for those workflows. And only for the new workflow we use the newly added `INVALIDATE_RKEY_ONLY` flag.
Maybe in the future we can improve other workflows, but this is definitely out of scope of this PR